src/campaign.cairo

```diff
@@ -116,20 +151,58 @@ mod TokengiverCampaign {
             self.withdrawal_balance.write(campaign_address, amount);
         }
 
+        fn withdraw(ref self: ContractState, campaign_address: ContractAddress, amount: u256) {
+            let campaign: Campaign = self.campaign.read(campaign_address);
+            let caller: ContractAddress = get_caller_address();
+
+            assert(caller == campaign.campaign_owner, NOT_CAMPAIGN_OWNER);
+
+            let available_balance: u256 = self.withdrawal_balance.read(campaign_address);
+            assert(amount <= available_balance, INSUFFICIENT_BALANCE);
+
+            self.withdrawal_balance.write(campaign_address, available_balance - amount);
+
+            // Transfer tokens
+            self.token.read().transfer_from(campaign_address, caller, amount);
```

Comment:
please use IECR20DISPATCHER to handle this transfer,

look at the work on donate function to see how it was implemented